File: VB/Form2.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
' ..

Namespace WindowsApplication1
	Partial Public Class Form2
		Inherits Form
		Public str As String = ""
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			str = Me.textBox1.Text
			Me.Close()
		End Sub

		Private Sub Form2_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Me.textBox1.Text = str
		End Sub
	End Class
End Namespace

File: CS/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1 {
    public partial class Form2 : Form {
        public string str = "";
        public Form2() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            str = this.textBox1.Text;
            this.Close();
        }

        private void Form2_Load(object sender, EventArgs e)
        {
            this.textBox1.Text = str;
        }
    }
}